hackathon_site/hardware/serializers.py

```diff
@@ -204,6 +208,16 @@ def create(self, validated_data):
         )
         new_order = None
         response_data = {"order_id": None, "hardware": [], "errors": []}
+
+        # The reason why doing this is because the id field stores the hardware object, django cannot translate hardware object into JSON. Therefore, loop has been used to call Profile seriliazer and assign the JSON file to the validated_data filed.
+        counter = 0
+        for item in validated_data["hardware"]:
+
+            temp = HardwareSerializer(item["id"]).data
+
+            validated_data["hardware"][counter] = temp
```

Comment:
Overwriting `validated_data` isn't a great idea, in case we try to use it for another purpose later on.

Also, the `request` field should also be storing the actual request that was made, it doesn't need to contain a fully serialized hardware instance - in other words, it should contain the data from `OrderCreateSerializer`, which you can get from `self.data`

Reply:
You'll also need to add tests to `OrderListViewPostTestCase` to make sure the request field is being built properly, since there aren't any direct serializer test for this - that would have caught that the request field is being built wrong (submit an order request and look at the order in the admin site to confirm this).

It looks like in most of the tests there, there is a `request_data` variable - just do `self.assertEqual(order.request, request_data)` in the test casses where the order is made successfully.

Reply:
> 

but this is written within `OrderCreateSerializer`. The serializer will call itself, doesn't it cause recursion?

Reply:
![image](https://user-images.githubusercontent.com/36619969/144698836-2881c7d4-38b0-46d2-8e1b-be91304f1e61.png)
 If we use self.data, we would need to also fetch the object every time, will this be not efficient?

Reply:
![image](https://user-images.githubusercontent.com/36619969/144699034-0d1c0d36-46ab-4ceb-9212-23b576e0f1f7.png)
![image](https://user-images.githubusercontent.com/36619969/144699089-a248931d-ce2a-40cc-9cbf-a0d1dcbabc5c.png)

If we use `self.data`, it will throw out this error. I tried to look into this problem, but I have no idea how this happend....

Reply:
I would need to see the stack trace to see where the error is coming from.

However, I can't reproduce the issue. I deleted the block of code on my end that uses the `counter` variable and such, made a post request, and everything worked as expected.

Reply:
If we remove that counter block,  I will get this error. That's the reason why I just overwritten the validated_data to make it become JSON able
![image](https://user-images.githubusercontent.com/36619969/146629333-201784b0-682a-479c-85ab-bb68030947d8.png)